tests/suites/test_suite_rsa.data

```diff
@@ -725,7 +725,7 @@ RSA parse public key - correct values, extra integer inside the SEQUENCE
 rsa_parse_pkcs1_key:1:"30818c028181009f091e6968b474f76f0e9c237c1d895996ae704b4f6d706acec8d2daac6209bf524aa3f658d0283adba1077f6cbe92e425dcde52290b239cade91be86c88425434986806e85734e159768f3dfea932baaa9409d25bace8ee9dce0cdde0903207299de575ae60feccf0daf82334ab83638539b0da74072f253acea8afc8e66bb70203010001020100":MBEDTLS_ERR_ASN1_LENGTH_MISMATCH
 
 RSA parse public key - correct values, extra integer outside the SEQUENCE
-rsa_parse_pkcs1_key:1:"308189028181009f091e6968b474f76f0e9c237c1d895996ae704b4f6d706acec8d2daac6209bf524aa3f658d0283adba1077f6cbe92e425dcde52290b239cade91be86c88425434986806e85734e159768f3dfea932baaa9409d25bace8ee9dce0cdde0903207299de575ae60feccf0daf82334ab83638539b0da74072f253acea8afc8e66bb70203010001":0
+rsa_parse_pkcs1_key:1:"308189028181009f091e6968b474f76f0e9c237c1d895996ae704b4f6d706acec8d2daac6209bf524aa3f658d0283adba1077f6cbe92e425dcde52290b239cade91be86c88425434986806e85734e159768f3dfea932baaa9409d25bace8ee9dce0cdde0903207299de575ae60feccf0daf82334ab83638539b0da74072f253acea8afc8e66bb70203010001020100":0
```

Comment:
Hold on: since there's extra data, the function should return an error.

And it looks like we're missing a similar test case for key pairs.

Reply:
> And it looks like we're missing a similar test case for key pairs.

It's on line 664:
`RSA parse private key - correct values, extra integer outside the SEQUENCE`

> Hold on: since there's extra data, the function should return an error.

This is basically the feedback I was waiting for after my [last comment](https://github.com/Mbed-TLS/mbedtls/pull/8740#issuecomment-1928901377):

> It makes more sense to me to do it this way because the user can provide a larger buffer which contains a valid key (priv or pub, it does not matter) and we are able to parse only the relevant data from there.

In my opinion, since the extra data is _after_ the sequence it could make sense to have a success here because we're not corrupting the main SEQUENCE (it's just some extra data after it). However if you think that this must still be an error I will update the code accordingly ;)

Reply:
I can't think of a direct security risk in accepting a key followed by arbitrary garbage, but I'm always uneasy accepting invalid inputs: it can often be a step in an attack in some weird circumstances that are hard to anticipate. So I would prefer to reject all invalid data unless there's a compelling reason.